vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -100,7 +100,10 @@ public void close() {
       JsonObject encodedDocument = encodeKeyWhenUseObjectId(document);
       filter.put(ID_FIELD, encodedDocument.getValue(ID_FIELD));
 
-      coll.replaceOne(wrap(filter), encodedDocument, convertCallback(resultHandler, result -> null));
+      com.mongodb.client.model.UpdateOptions updateOptions = new com.mongodb.client.model.UpdateOptions()
+              .upsert(true);
+
+      coll.replaceOne(wrap(filter), encodedDocument, updateOptions, convertCallback(resultHandler, result -> null));
```

Comment:
I'm going to bite my tongue about coll vs collection naming (I know, I know, we're following what's previously there) :-)